sycl/doc/extensions/proposed/sycl_ext_intel_cslice.asciidoc

```diff
@@ -0,0 +1,273 @@
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+:multi-CCS: https://github.com/intel/compute-runtime/blob/master/level_zero/doc/experimental_extensions/MULTI_CCS_MODES.md
+
+Some Intel GPU devices can be partitioned at a granularity of "cslice" (compute
+slice), which is a smaller granularity than "tile".  This form of partitioning
+is not currently enabled by default, so it is considered an advanced feature
+which most applications are not expected to use.  This extension provides a way
+for these advanced applications to partition a device by cslice when it is
+enabled in the device driver.
+
+Unlike "tile" partitions, a cslice partition does not have any different cache
+affinity from its sibling cslice partitions.  Therefore, this extension does
+not expose this type of partitioning through
+`info::partition_property::partition_by_affinity_domain`.  Instead, it adds a
+new partitioning type
+`info::partition_property::ext_intel_partition_by_cslice`.
+
+The only Intel GPU device that currently supports this type of partitioning is
+PVC, and this support is only available when the device driver is configured in
+{multi-CCS}[multi-CCS] mode.  See that documentation for instructions on how to
+enable this mode and for other important information.  Currently, it is only
+possible to partition a device by cslice if the driver is in "2 CCS Mode" or
+"4 CCS Mode".  When in 2 CCS Mode, a tile can be partitioned into two cslice
+sub-devices.  When in 4 CCS Mode, a tile can be partitioned into four cslice
+sub-devices.
+
+This type of partitioning is currently supported only at the "tile" level.
+Therefore, a device must first be partitioned into per-tile sub-devices via
+`partition_by_affinity_domain`, and then each of the resulting sub-devices can
+be further partitioned by `ext_intel_partition_by_cslice`.
+
+It is important to understand that the device driver virtualizes work
+submission to the cslice sub-devices.  This virtualization happens only between
+processes, and not within a single process.  For example, consider a single
+process that constructs two SYCL queues on cslice sub-device #0.  Kernels
+submitted to these two queues are guaranteed to conflict, both using the same
+set of execution units.  Therefore, if a single process wants to explicitly
+submit kernels to cslice sub-devices and it wants to avoid conflict, it should
+create queues on different sub-devices.  By contrast, consider an example where
+two separate processes create a SYCL queue on cslice sub-device #0.  In this
+case, the device driver virtualizes access to this cslice, and kernels
+submitted from the first process may run on different execution units than
+kernels submitted from the second process.  In this second case, the device
+driver binds the process's requested cslice to a physical cslice according to
+the overall system load.
```

Comment:
to keep in mind: this is only true for PVC. For other GPUs, even when submitting to the different CCSs from different processes, they might end up conflicting on EU usage, since they would be using EUs from all available c-slices and there's no c-slice affinity.